### PR TITLE
Implement GET /apis/{apiId}/subscribers

### PR DESCRIPTION
**Pull request by phiz71**
## Issue

https://gravitee.atlassian.net/browse/APIM-1227

## Description

Add a paginated resource to list subscribers of an API.

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-fdjclibnbf.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 1]
```diff
@@ -795,6 +795,34 @@ paths:
...
+    # API Subscriptions
+    /environments/{envId}/apis/{apiId}/subscribers:
+        parameters:
+            - $ref: "#/components/parameters/envIdParam"
+            - $ref: "#/components/parameters/apiIdParam"
+        get:
+            parameters:
+                - name: query
+                  description: Query to filter subscribers by name.
+                  in: query
```
Comment (jhaeyaert) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:809]: Could be rename "name" to make it clearer

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 2]
```diff
@@ -795,6 +795,34 @@ paths:
                 default:
                     $ref: "#/components/responses/Error"
 
+    # API Subscriptions
```
Comment (jhaeyaert) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:797]: Api Subscribers ?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiResource_GetApiSubscribersTest.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    SubscriptionService subscriptionService;
+
+    @Before
+    public void init() throws TechnicalException {
+        super.init();
+        Mockito.reset(applicationService, subscriptionService);
+    }
+
+    @Test
+    public void should_return_empty_page_if_no_subscribtions() {
```
Comment (jhaeyaert) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiResource_GetApiSubscribersTest.java]: if_no_subscribers ? (I know that we are listing applications having subscriptions but I found it more obvious to name it

**Review by jhaeyaert — CHANGES_REQUESTED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml (hunk 2 above)
Comment (phiz71) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml:797]: The `subscriber` notion is very close to subscriptions. And this comment helps to group resources by tag in the yaml file 🙂

**Review by jhaeyaert — APPROVED** [review]

**Pull request merged**